src/common/helpers.ts

```diff
@@ -158,6 +151,23 @@ export function getResponse(op: Operation): SchemaResponse | undefined {
   return with200;
 }
 
+// returns true if the operation returns multiple response types
+export function isMultiRespOperation(op: Operation): boolean {
+  // treat LROs as single-response ops
+  if (!op.responses || op.responses.length === 1 || isLROOperation(op)) {
+    return false;
+  }
+  // count the number of distinct schemas returned by this operation
+  const schemaResponses = new Array<SchemaResponse>();
+  for (const response of values(op.responses)) {
+    // perform the comparison by name as some responses have different objects for the same underlying response type
```

Comment:
The check below is making sure that if two schema responses are using the same schema, they will share they underlying response envelope right?

Reply:
Yes but indirectly as a result of multi-response operations having the same response envelope with a `Value interface{}` field.